api_client.py

```diff
@@ -421,19 +438,20 @@ def get_tenants_from_sophos(self):
                     tenant_data = self.get_partner_organization_tenants(
                         whoami_response, access_token
                     )
-                    tenant_data["access_token"] = access_token
                 else:
-                    if (
-                        self.config.tenant_id != ""
-                        and self.config.tenant_id
-                        != whoami_response["id"]
-                    ):
-                        raise Exception(
-                            "Configuration file mention tenant id not matched with whoami data tenant id"
-                        )
-                    else:
-                        tenant_data = whoami_response
-                        tenant_data["access_token"] = access_token
+                    tenant_data = {
+                        "items": [ whoami_response ],
+                        "access_token": access_token
```

Comment:
It's a bit hacky to stuff the access_token in here. It might be better to use multiple return values or to pull the authentication out of this method into a higher scope.